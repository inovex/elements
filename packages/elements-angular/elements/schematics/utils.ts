import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
