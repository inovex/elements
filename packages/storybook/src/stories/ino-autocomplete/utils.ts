import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
