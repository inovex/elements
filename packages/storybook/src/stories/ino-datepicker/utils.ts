import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
