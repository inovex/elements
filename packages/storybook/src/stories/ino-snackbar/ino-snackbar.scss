.story-ino-snackbar {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 24px;

  ino-button:not(:last-of-type) {
    margin-bottom: 32px;
  }
}
