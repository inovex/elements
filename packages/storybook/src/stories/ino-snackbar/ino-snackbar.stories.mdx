import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-snackbar/readme.md";

<Meta title="Notification/ino-snackbar/Documentation" />

<Description markdown={readme} />
