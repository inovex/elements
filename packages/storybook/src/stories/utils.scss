/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
