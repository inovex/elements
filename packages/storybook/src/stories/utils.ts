import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
