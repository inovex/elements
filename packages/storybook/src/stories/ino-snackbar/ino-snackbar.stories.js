import { number, select, text } from '@storybook/addon-knobs';
import withStencilReadme from '_local-storybookcore/with-stencil-readme';
import componentReadme from '_local-elements/src/components/ino-snackbar/readme.md';
import './ino-snackbar.scss';

function subscribeToComponentEvents() {
  // == event block
  const eventHandler = function (e) {
    const target = e.target;

    if (target.tagName !== 'INO-BUTTON') {
      return;
    }

    const templates = Array.from(document.getElementsByTagName('template'));
    const templateWithId = templates.find((template) => template.id === target.id);

    document.body.appendChild(templateWithId.content.cloneNode(true));
  };

  document.addEventListener('click', eventHandler);
  // == event block

  // unsubscribe function will be called by Storybook
  return () => {
    document.removeEventListener('click', eventHandler);
  };
}

const sampleText =
  'Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua.';

export default {
  title: 'Notification/<ino-snackbar>',

  decorators: [
    withStencilReadme(componentReadme),
    (story) => {
      subscribeToComponentEvents();
      return story();
    },
  ],
};

export const DefaultUsage = () => /*html*/ `
    <div class="story-message-box">
      <div class="flex-parent-center">
          <h4>Customizable Snackbar</h4>
          <style>
            ino-snackbar.customizable-snackbar {
              --ino-snackbar-warning-border-color: ${text(
                '--ino-snackbar-warning-border-color',
                '#ffde03',
                'Custom Properties'
              )};
              --ino-snackbar-error-border-color: ${text(
                '--ino-snackbar-error-border-color',
                '#eb003b',
                'Custom Properties'
              )};
            }
          </style>
          <ino-button id="snackbar-temp">Show Snackbar</ino-button>
          <template id="snackbar-temp">
            <ino-snackbar
                class="customizable-snackbar"
                id="custom-snackbar"
                ino-message="${text('ino-message', sampleText)}"
                ino-action-text="${text('ino-action-text', 'Anlegen')}"
                ino-timeout="${number('ino-timeout', 5000)}"
                ino-type="${select('ino-type', ['primary', 'error', 'warning'], 'primary')}"
                ino-alignment="${select(
                  'ino-alignment',
                  ['center', 'leading', 'trailing'],
                  'center'
                )}" />
          </template>

        <h4>Variants</h4>
        <div class="snackbar-variants">
          <ino-button id="snackbar-with-action-temp">Show Snackbar (With Action)</ino-button>
          <template id="snackbar-with-action-temp">
              <ino-snackbar
                  ino-message="${sampleText}"
                  ino-action-text="Action"
                  ino-alignment="left" />
            </template>
          <ino-button id="snackbar-no-action-temp">Show Snackbar (No Action)</ino-button>
          <template id="snackbar-no-action-temp">
            <ino-snackbar
                ino-message="${sampleText}"
                ino-alignment="right" />
          </template>
        </div>
        <h4>Snackbar Types</h4>
        <div class="snackbar-variants">
          <ino-button id="snackbar--warning">Show Snackbar (warning)</ino-button>
          <template id="snackbar--warning">
            <ino-snackbar
                ino-message="${sampleText}"
                ino-action-text="Action"
                ino-type="warning"/>
          </template>
          <ino-button id="snackbar--error">Show Snackbar (error)</ino-button>
          <template id="snackbar--error">
            <ino-snackbar
                ino-message="${sampleText}"
                ino-action-text="Action"
                ino-type="error" />
          </template>
        </div>
        <h4>Alignments</h4>
        <div class="snackbar-variants">
            <ino-button id="left-align">Left</ino-button>
            <template id="left-align">
              <ino-snackbar
                  ino-message="${sampleText}"
                  ino-action-text="Anlegen"
                  ino-alignment="left"
                  >
              </ino-snackbar>
            </template>
            <ino-button id="center-align">Center</ino-button>
            <template id="center-align">
              <ino-snackbar
                  ino-message="${sampleText}"
                  ino-action-text="Anlegen"
                  >
              </ino-snackbar>
            </template>
            <ino-button id="right-align">Right</ino-button>
            <template id="right-align">
              <ino-snackbar
                  ino-message="${sampleText}"
                  ino-action-text="Anlegen"
                  ino-alignment="right"
                  >
              </ino-snackbar>
            </template>
            </div>
        </div>
    </div>
  `;
