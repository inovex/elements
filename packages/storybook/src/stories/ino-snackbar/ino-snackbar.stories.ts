import { Components } from '@inovex.de/elements';
import { html } from 'lit-html';
import { useEffect } from '@storybook/client-api';
import { defaultDecorator } from '../utils';
import { Story } from '@storybook/web-components';
import './ino-snackbar.scss';

const sampleText = 'User successfully updated!';

export default {
  title: 'Notification/ino-snackbar',
  component: 'ino-snackbar',
  decorators: [
    (s) => defaultDecorator(s, 'story-ino-snackbar'),
    (story) => {
      const btnClickHandler = (e) => {
        if (!e.target.classList.contains('snackbar-trigger')) {
          return;
        }
        const templates = Array.from(document.getElementsByTagName('template'));
        const templateWithId = templates.find(
          (template) => template.id === e.target.dataset.templateId
        );
        document.body.appendChild(templateWithId.content.cloneNode(true));
      };
      useEffect(() => {
        document.addEventListener('click', btnClickHandler);
        return () => document.removeEventListener('click', btnClickHandler);
      });
      return story();
    },
  ],
};

export const Playground: Story<Components.InoSnackbar> = (args) => html`
  <ino-button class="snackbar-trigger" data-template-id="snackbar-default"
    >Show Snackbar</ino-button
  >
  <template id="snackbar-default">
    <ino-snackbar
      action-text="${args.actionText}"
      alignment="${args.alignment}"
      message="${args.message}"
      timeout="${args.timeout}"
      type="${args.type}"
    >
    </ino-snackbar>
  </template>
`;
Playground.args = {
  actionText: 'Anlegen',
  alignment: 'center',
  message: sampleText,
  timeout: -1,
  type: 'primary',
};
Playground.argTypes = {
  alignment: {
    control: {
      type: 'select',
    },
    options: ['left', 'top', 'center'],
  },
  type: {
    control: {
      type: 'select',
    },
    options: ['primary', 'warning', 'error'],
  },
};

export const Variants = () => html`
  <ino-button class="snackbar-trigger" data-template-id="snackbar-with-action"
    >Show Snackbar (With Action)</ino-button
  >
  <template id="snackbar-with-action">
    <ino-snackbar message="${sampleText}" action-text="Anlegen"></ino-snackbar>
  </template>

  <ino-button class="snackbar-trigger" data-template-id="snackbar-no-action"
    >Show Snackbar (No Action)</ino-button
  >
  <template id="snackbar-no-action">
    <ino-snackbar message="${sampleText}"></ino-snackbar>
  </template>

  <ino-button class="snackbar-trigger" data-template-id="snackbar-no-timeout"
    >Show Snackbar (Without timeout)</ino-button
  >
  <template id="snackbar-no-timeout">
    <ino-snackbar message="${sampleText}" timeout="-1"></ino-snackbar>
  </template>
`;

export const Types = () => html`
  <ino-button class="snackbar-trigger" data-template-id="snackbar-warning"
    >Show Snackbar (warning)</ino-button
  >
  <template id="snackbar-warning">
    <ino-snackbar message="${sampleText}" type="warning"></ino-snackbar>
  </template>

  <ino-button class="snackbar-trigger" data-template-id="snackbar-error"
    >Show Snackbar (error)</ino-button
  >
  <template id="snackbar-error">
    <ino-snackbar message="${sampleText}" type="error"></ino-snackbar>
  </template>
`;

export const Alignments = () => html`
  <ino-button class="snackbar-trigger" data-template-id="snackbar-left-align"
    >Left</ino-button
  >
  <template id="snackbar-left-align">
    <ino-snackbar
      message="${sampleText}"
      action-text="Anlegen"
      alignment="left"
    ></ino-snackbar>
  </template>

  <ino-button class="snackbar-trigger" data-template-id="snackbar-center-align"
    >Center</ino-button
  >
  <template id="snackbar-center-align">
    <ino-snackbar message="${sampleText}" action-text="Anlegen"></ino-snackbar>
  </template>

  <ino-button class="snackbar-trigger" data-template-id="snackbar-right-align"
    >Right</ino-button
  >
  <template id="snackbar-right-align">
    <ino-snackbar
      message="${sampleText}"
      action-text="Anlegen"
      alignment="right"
    ></ino-snackbar>
  </template>
`;

// TODO CSS Variable
/*
ino-snackbar.customizable-snackbar {
  --ino-snackbar-warning-border-color: ${text(
    '--ino-snackbar-warning-border-color',
    '#ffde03',
    'Custom Properties'
  )};
  --ino-snackbar-error-border-color: ${text(
    '--ino-snackbar-error-border-color',
    '#eb003b',
    'Custom Properties'
  )};
}
*/
