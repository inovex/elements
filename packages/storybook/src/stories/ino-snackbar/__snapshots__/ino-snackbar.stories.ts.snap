// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification/ino-snackbar ActionText smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-snackbar">
    <ino-button class="snackbar-trigger hydrated"
                data-template-id="snackbar-actionText"
    >
      <button class="button button__variant--filled"
              type="button"
      >
        <span class="button__label">
          Show Snackbar
        </span>
      </button>
    </ino-button>
    <template id="snackbar-actionText">
      <ino-snackbar id="snackbar-actionText"
                    action-text="Show"
                    timeout="-1"
                    type="info"
                    stay-visible-on-hover="false"
      >
        You received a new message.
      </ino-snackbar>
    </template>
  </div>
</div>
`;

exports[`Notification/ino-snackbar Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-snackbar">
    <ino-button class="snackbar-trigger hydrated"
                data-template-id="snackbar-default"
    >
      <button class="button button__variant--filled"
              type="button"
      >
        <span class="button__label">
          Show Snackbar
        </span>
      </button>
    </ino-button>
    <template id="snackbar-default">
      <ino-snackbar id="snackbar-default"
                    action-text="Some Action"
                    timeout="-1"
                    type="info"
                    stay-visible-on-hover="false"
      >
        This is a message
      </ino-snackbar>
    </template>
  </div>
</div>
`;

exports[`Notification/ino-snackbar StayVisibleOnHover smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-snackbar">
    <ino-button class="snackbar-trigger hydrated"
                data-template-id="snackbar-stayVisibleOnHover"
    >
      <button class="button button__variant--filled"
              type="button"
      >
        <span class="button__label">
          Show Snackbar
        </span>
      </button>
    </ino-button>
    <template id="snackbar-stayVisibleOnHover">
      <ino-snackbar id="snackbar-stayVisibleOnHover"
                    action-text="Some Action"
                    timeout="5000"
                    type="info"
                    stay-visible-on-hover="true"
      >
        This snackbar stays visible on hover otherwise it will disappear in 5s
      </ino-snackbar>
    </template>
  </div>
</div>
`;

exports[`Notification/ino-snackbar Timeout smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-snackbar">
    <ino-button class="snackbar-trigger hydrated"
                data-template-id="snackbar-timeout"
    >
      <button class="button button__variant--filled"
              type="button"
      >
        <span class="button__label">
          Show Snackbar
        </span>
      </button>
    </ino-button>
    <template id="snackbar-timeout">
      <ino-snackbar id="snackbar-timeout"
                    action-text="Some Action"
                    timeout="5000"
                    type="info"
                    stay-visible-on-hover="false"
      >
        This snackbar will disappear in 5s
      </ino-snackbar>
    </template>
  </div>
</div>
`;

exports[`Notification/ino-snackbar Types smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-snackbar">
    <ino-button class="snackbar-trigger hydrated"
                data-template-id="snackbar-type"
    >
      <button class="button button__variant--filled"
              type="button"
      >
        <span class="button__label">
          Show Snackbar
        </span>
      </button>
    </ino-button>
    <template id="snackbar-type">
      <ino-snackbar id="snackbar-type"
                    action-text="Undo"
                    timeout="-1"
                    type="success"
                    stay-visible-on-hover="false"
      >
        User successfully updated!
      </ino-snackbar>
    </template>
  </div>
</div>
`;
